funnel/models/event.py

```diff
@@ -52,6 +53,25 @@ class Event(BaseMixin, db.Model):
     ticket_types = db.relationship("TicketType", secondary=event_ticket_type, lazy='dynamic')
     participants = association_proxy('attendees', 'participant')
 
+    @classmethod
+    def get_or_create(cls, name, space):
+        event = cls.query.filter_by(name=name, proposal_space=space).first()
+        if not event:
```

Comment:
Change to `if not event and create:`

Also, pass `create=True` when calling it.